src/OrchardCore.Modules/OrchardCore.Contents/Recipes/ContentStep.cs

```diff
@@ -36,24 +37,52 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                 var contentItem = token.ToObject<ContentItem>();
                 var modifiedUtc = contentItem.ModifiedUtc;
                 var publishedUtc = contentItem.PublishedUtc;
+                var isPublished = contentItem.Published;
+                var isLatest = contentItem.Latest;
+
                 var existing = await _contentManager.GetVersionAsync(contentItem.ContentItemVersionId);
-                
+
                 if (existing == null)
                 {
                     // Initializes the Id as it could be interpreted as an updated object when added back to YesSql
                     contentItem.Id = 0;
                     await _contentManager.CreateAsync(contentItem);
-                    
+
                     // Overwrite ModifiedUtc & PublishedUtc values that handlers have changes
                     // Should not be necessary if IContentManager had an Import method
                     contentItem.ModifiedUtc = modifiedUtc;
                     contentItem.PublishedUtc = publishedUtc;
-                }
-                else
-                {
-                    // Replaces the id to force the current item to be updated
-                    existing.Id = contentItem.Id;
-                    _session.Save(existing);
+
```

Comment:
removed this as it makes no sense to save an item that has just been retrieved, and not mutated.
If we did want to do mutatations, this would be the place.
Removing this saves around 10 sql calls to indexes etc, to update / delete / create indexes, that have not changed.

Reply:
You sure because _session.Save is a no op method?

Reply:
Ok here what it does is that it will save the existing content item if nothing is done afterward. So if it doesn't find any relatedContentItems it will still call a _session.Save(existing)

So we have

contentItem = passed from the step context
existing = retrieved from DB and specific version (so could be draft or published)
relatedContentItem = any contentItem (drafted or published) related with the contentItem passed from the step context.

Then you iterate over these relatedContentItem to save each of them and you set these as Published or Draft based on contentItem.Published or ContentItem.Latest.

Reply:
Ok we can remove because it will still be called (I assume)